From e75e77b6cb9ba34cb4ed6060b06fe37e362f1451 Mon Sep 17 00:00:00 2001
From: Yang Fan F <fan.f.yang@intel.com>
Date: Fri, 5 Feb 2016 16:13:50 +0800
Subject: [PATCH 01/24] atomisp2: css fix one kw issue

Change-Id: I8ebaa4d10af9af17443490842fea9df805a98e6c
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18671
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/99806
Validation-Android: Wei Tao <wei.tao@intel.com>
Reviewed-by: Manoj Dawarwadikar <manoj.dawarwadikar@intel.com>
Reviewed-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-by: Bingbu Cao <bingbu.cao@intel.com>
Approver: Bingbu Cao <bingbu.cao@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Reviewed-on: https://android.intel.com:443/466890
---
 .../camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c        | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c
index f256211..fe7ff61 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_shading.c
@@ -106,6 +106,8 @@ crop_and_interpolate(unsigned int cropped_width,
 	out_cell_size = CEIL_DIV(padded_width, out_table->width - 1);
 	in_cell_size  = CEIL_DIV(sensor_width, table_width - 1);
 
+	assert(in_cell_size != 0);
+
 	out_start_col = ((int)sensor_width - (int)cropped_width)/2 - left_padding;
 	out_start_row = ((int)sensor_height - (int)cropped_height)/2 - top_padding;
 	table_cell_w = (int)((table_width-1) * in_cell_size);
-- 
1.9.1

